src/AWSS3.jl

```diff
@@ -62,22 +62,38 @@ from `path` in `bucket`.
                 (common if object was recently created).
 - `raw=false`:  return response as `Vector{UInt8}`
                 (by default return type depends on `Content-Type` header).
+- `byte_range=nothing`:  given an iterator of `(start_byte, end_byte)` gets only
+    the range of bytes of the object from `start_byte` to `end_byte`.  For example,
+    `byte_range=1:4` gets bytes 1 to 4 inclusive.  Arguments should use the Julia convention
+    of 1-based indexing.
 - `header::Dict{String,String}`: pass in an HTTP header to the request.
 
-For example, to get a `range` of bytes instead of the whole object, do:
+As an example of how to set custom HTTP headers, the below is equivalent to
+`s3_get(aws, bucket, path; byte_range=range)`.
 
 `s3_get(aws, bucket, path; headers=Dict{String,String}("Range" => "bytes=\$(first(range)-1)-\$(last(range)-1)"))`
 """
-function s3_get(aws::AbstractAWSConfig, bucket, path; version="", retry=true, raw=false, headers=Dict{String, Any}(), kwargs...)
+function s3_get(aws::AbstractAWSConfig, bucket, path; version="", retry=true,
+                byte_range=nothing, raw=false,
+                headers::AbstractDict{<:AbstractString,<:Any}=Dict{String, Any}(),
+                return_stream::Bool=false, kwargs...)
```

Comment:
Can we run JuliaFormatter w/ BlueStyle on this, and specify a type for `byte_range`?

Reply:
Personally I strongly prefer to use appropriate type assertions on function arguments in most places, but I was trying not to break to starkly with the conventions in this package.  The assertion on `headers` was added because I was afraid that strange things can happen if it was passed a different type, given the code below.

This should be resolved now.